tests/chocolatey-tests/choco-upgrade.Tests.ps1

```diff
@@ -0,0 +1,122 @@
+Import-Module helpers/common-helpers
+
+Describe "choco upgrade" -Tag Chocolatey, UpgradeCommand {
+    BeforeAll {
+        Initialize-ChocolateyTestInstall
+
+        New-ChocolateyInstallSnapshot
+    }
+
+    AfterAll {
+        Remove-ChocolateyTestInstall
+    }
+
+    Context "Upgrading a side-by-side Package (non-existing)" {
+        BeforeAll {
+            Restore-ChocolateyInstallSnapshot
+
+            $PackageUnderTest = "installpackage"
+
+            $Output = Invoke-Choco upgrade $PackageUnderTest --confirm --allowmultipleversions
+        }
+
+        It "Exits with Success (0)" {
+            $Output.ExitCode | Should -Be 0
+        }
+
+        It "Installed a package to the lib directory" {
+            "$env:ChocolateyInstall\lib\$($PackageUnderTest).1.0.0" | Should -Exist
+        }
+
+        It "Installs the expected version of the package" {
+            "$env:ChocolateyInstall\lib\$($PackageUnderTest).1.0.0\$($PackageUnderTest).1.0.0.nuspec" | Should -Exist
+            [xml]$XML = Get-Content "$env:ChocolateyInstall\lib\$($PackageUnderTest).1.0.0\$($PackageUnderTest).1.0.0.nuspec"
+            $XML.package.metadata.version | Should -Be "1.0.0"
+        }
+
+        It "Outputs a warning message about side by side installs are deprecated" {
+            $Output.Lines | Should -Contain "Upgrading the same package with multiple versions is deprecated and will be removed in v2.0.0." -Because $Output.String
+        }
+
+        It "Outputs a message indicating that it upgraded the package successfully" {
+            $Output.Lines | Should -Contain "Chocolatey upgraded 1/1 packages." -Because $Output.String
+        }
+    }
+
+    Context "Switching a normal Package to a side-by-side Package" {

```

Comment:
Could we skip this if version is below 1.2.0-alpha? That will allow these tests to be bypassed with current stable builds of Chocolatey.